weavertest/init.go

```diff
@@ -69,6 +69,7 @@ type testMainInterface interface{}
 //	func(ComponentType, ComponentType2)
 //	...
 func Run(t testing.TB, opts Options, testBody any) {
+	_, isBench := t.(interface{ Elapsed() time.Duration })
```

Comment:
I'm just curious why you didn't do: `_, isBench := t.(*testing.B)`

Reply:
I was worried about the case where the caller had wrapped a *testing.B into their own type that implemented the same interface as *testing.B.

Perhaps it is better to just do the simple thing for now.